docs/narr.rst

```diff
@@ -571,16 +571,84 @@ There is a ``condition`` attribute in the
 "http://namespaces.zope.org/zcml" namespace which is honored on all
 elements in ZCML.  The value of the attribute is an expression
 which is used to determine if that element and its descendents are
-used.  If the condition is true, processing continues normally,
+used.  If the condition evaluates to true, processing continues normally;
 otherwise that element and its descendents are ignored.
 
-Currently the expression is always of the form "have featurename", and it
-checks for the presence of a ``<meta:provides feature="featurename" />``.
+The expression is always of the form "*verb* *arguments*". There are
+four verbs that are supported:
+
+- have
+- not-have
+- installed
+- not-installed
+
+Note that logical "or" conditions can be achieved by using both the
+positive verb and its not-prefixed counterpart on sibling elements
+with the same arguments. Logical "and" conditions can be achieved by
+nesting elements. To group arbitrary directives under a condition,
+nest them under a new ``<configure>`` element.
+
+.. seealso:: :meth:`zope.configuration.xmlconfig.ConfigurationHandler.evaluateCondition`
+
+   For documentation and examples of the XML implementation of conditions.
+
+Features
+--------
+
+The verbs ``have`` and ``not-have`` test for the presence or absense
+of features in the configuration context. The argument is a single
+feature name. Features can be established in Python code using
+:meth:`~zope.configuration.config.ConfigurationContext.provideFeature`,
+or they can be set during ZCML processing with the ``meta:provides``
+directive (e.g., ``<meta:provides feature="featurename" />``.)
+
+A popular use of features is to enable certain configuration only
+during development with a feature called ``devmode``. (This also
+demonstrates applying a condition to a single directive.)
+
+.. code-block:: xml
+
+   <configure
+      xmlns="http://namespaces.zope.org/zope"
+      xmlns:zcml="http://namespaces.zope.org/zcml" >
+      <include package="." file="devconfig.zcml" zcml:condition="have devmode" />
+   </configure>
+
+Module Availability
+-------------------
+
+The verbs ``installed`` and ``not-installed`` test whether a Python
+module can be imported successfully or not. They can be used to enable
+optional configuration if a package is present, or enable fallback
+functionality if it isn't. (This also demonstrates grouping directives
+under a new ``<configure>`` element, and establishing a logical "or".)
```

Comment:
I don't think this is a logical "or", more of an if-else.

To me "logical or" would be something like "feature devmode or feature this-specific-thing".

Reply:
Good point, I'll clarify.